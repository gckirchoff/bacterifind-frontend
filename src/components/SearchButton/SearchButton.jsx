import React from 'react';
import './SearchButton.scss';

const SearchButton = ({ finalCode, handleSearchClick }) => {
  return (
    <button className="search-btn" onClick={handleSearchClick}>
      <span className=" search-btn__visible">{finalCode}</span>
      <span className="search-btn__invisible">Search!</span>
    </button>
  );
};

export default SearchButton;
