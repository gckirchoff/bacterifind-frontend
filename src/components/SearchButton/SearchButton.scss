// .search-button {
//   font-size: 1rem;
//   font-weight: 300;
//   color: white;
//   text-transform: uppercase;
//   border: none;
//   border-radius: 100px;
//   background-image: linear-gradient(to right, red, blue);
//   position: relative;
//   overflow: hidden;
//   cursor: pointer;

//   & > * {
//     display: inline-block;
//     height: 100%;
//     width: 100%;
//     transition: all 0.2s;
//   }

//   .visible {
//     padding: 2rem 7.5rem;
//   }

//   .invisible {
//     position: absolute;
//     left: 0;
//     top: -100%;
//     padding: 2rem 0;
//   }

//   &:hover {
//     background-image: linear-gradient(to left, red, blue);
//   }

//   &:hover .visible {
//     transform: translateY(100%);
//   }

//   &:hover .invisible {
//     top: 0;
//   }

//   &:focus {
//     outline: none;
//     animation: pulse 1s infinite;
//   }
// }

.search-btn {
  font-size: 1.3rem;
  font-weight: 300;
  color: white;
  border-radius: 100px;
  border: none;
  background-image: linear-gradient(
    to left,
    rgb(66, 206, 248),
    rgb(10, 152, 196)
  );
  position: relative;
  overflow: hidden;
  cursor: pointer;
  box-sizing: border-box;
  margin-top: 1.75rem;

  & > * {
    display: inline-block;
    height: 100%;
    transition: all 0.2s;
  }

  &__visible {
    padding: 0.7rem 1.2rem;
  }

  &__invisible {
    position: absolute;
    left: 0;
    top: -200%;
    padding: 2rem 0;
    width: 100%;
  }

  &:hover {
    background-image: linear-gradient(
      to right,
      rgb(66, 206, 248),
      rgb(10, 152, 196)
    );
  }

  &:hover &__visible {
    transform: translateY(100%);
  }

  &:hover &__invisible {
    top: -45%;
  }

  &:focus {
    outline: none;
  }
}
