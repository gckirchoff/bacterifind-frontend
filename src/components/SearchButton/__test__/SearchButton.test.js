import React from 'react';
import ReactDOM from 'react-dom';
import SearchButton from './../SearchButton';

import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import renderer from 'react-test-renderer';

afterEach(cleanup);
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<SearchButton />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders button correctly', () => {
  const { getByTestId } = render(<SearchButton finalCode={4000} />);
  expect(getByTestId('search-button')).toHaveTextContent('Search 4000');
});

it('renders button with alternative text', () => {
  const { getByTestId } = render(<SearchButton finalCode={54023} />);
  expect(getByTestId('search-button')).toHaveTextContent('Search 54023');
});

it('matches snapshot', () => {
  const tree = renderer.create(<SearchButton finalCode={40002} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('matches snapshot 2', () => {
  const tree = renderer.create(<SearchButton finalCode={40003} />).toJSON();
  expect(tree).toMatchSnapshot();
});
