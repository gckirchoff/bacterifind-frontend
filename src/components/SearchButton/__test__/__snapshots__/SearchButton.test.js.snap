// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="search-btn"
  data-testid="search-button"
>
  Search 
  40002
</button>
`;

exports[`matches snapshot 2 1`] = `
<button
  className="search-btn"
  data-testid="search-button"
>
  Search 
  40003
</button>
`;
